cryostatio/cryostat3
# feat(recordings): implement JFR snapshot

# Welcome to Cryostat3! 👋
## Before contributing, make sure you have:

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Fixes: #177

## Description of the change:
Reimplements JFR snapshot API v1 and v2, plus adds v3. These are all quite similar, just the response body format and status codes vary slightly.

## How to manually test:
1. `./mvnw clean verify ; podman image prune -f`
2. `./smoketest.bash -OXtb` and wait for things to come up
3. Select a target
4. Recordings > Create > Snapshot > Create . This should fail and display a warning notification message about there being no source data to snapshot.
5. Recordings > Create > Custom > create a new recording
6. Repeat step 4, this time it should succeed.
